### PR TITLE
Add support for vim command line options, for #858

### PR DESCRIPTION
**Pull request by Yohannfra**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Add vim command line options 
**What is the current behavior? (You can also link to an open issue here)**
No command line options
**What is the new behavior (if this is a feature change)?**
Command line options

### CODE REVIEW

File: src/vim.ts [hunk 1]
```diff
@@ -5,6 +5,260 @@ const completionSpec: Fig.Spec = {
...
+    },
+    {
+      name: "-r (with file)",
+      description: "Recover crashed session",
+      args: {
+        name: "filename",
+        isOptional: false,
+        template: "filepaths",
+      },
+    },
```
Comment (mschrage) [inline src/vim.ts:101]: The `name` field must only contain the flag name and nothing else. Please combine the two `-r` flags.

File: src/vim.ts [hunk 2]
```diff
@@ -5,6 +5,260 @@ const completionSpec: Fig.Spec = {
...
+        name: "command",
+        isOptional: false,
+      },
+    },
+    {
+      name: "-c",
+      description: "Execute <command> after loading the first file",
+      args: {
+        name: "command",
+        isOptional: false,
```
Comment (mschrage) [inline src/vim.ts:189]: Add `isCommand: true` here so that Fig will continue to provide suggestions :)

File: src/vim.ts [hunk 3]
```diff
@@ -5,6 +5,260 @@ const completionSpec: Fig.Spec = {
...
+        template: "filepaths",
+      },
+    },
+    {
+      name: "-i",
+      description: "Use <viminfo> instead of .viminfo",
+      args: {
+        name: "viminfo",
+        isOptional: false,
```
Comment (mschrage) [inline src/vim.ts:246]: Should this have `template: "filepaths"`?

**Review by mschrage — CHANGES_REQUESTED** [review]
Thanks so much for fleshing out the `vim` spec @Yohannfra. There are a few minor changes before we can get this merged in.

File: src/vim.ts (hunk 3 above)
Comment (Yohannfra) [reply src/vim.ts:244]: yes thanks

File: src/vim.ts [hunk 4]
```diff
@@ -5,6 +5,259 @@ const completionSpec: Fig.Spec = {
...
+      name: "-n",
+      description: "No swap file, use memory only",
+    },
+    {
+      name: "-r",
+      description:
+        "Recover crashed session if filename is specified, otherwise list swap files and exit",
+      args: {
+        name: "filename",
+        isOptional: false,
```
Comment (mschrage) [inline src/vim.ts:95]: `isOptional` should be `true` here, right?

Comment (Yohannfra) [reply src/vim.ts:95]: yes oops! thanks you

**Review by mschrage — APPROVED** [review]

**Pull request merged**